Package rfsm.2.0

jserot opened the pull request:
### `rfsm.2.0`
A toolset for describing and simulating StateChart-like state diagrams



---
* Homepage: http://github.com/jserot/rfsm
* Source repo: git+https://github.com/jserot/rfsm.git
* Bug tracker: jocelyn.serot@uca.fr

---
:camel: Pull-request generated by opam-publish v2.2.0

jserot commented:
`opam-ci` fails here:
```
# File "src/old/bin/syntax.ml", line 19, characters 11-22:
# 19 |   te_desc: Type_expr.t;
```
but there's no such file in the source tree (nor in the corresponding release in the GH repo) !
`dune build` and `opam install` works fine on my machine.
I suspect `opam-ci` is not using the correct source tree.

jserot commented:
Hmm.
Here's the content of 'rfsm.opam` file on my local machine:
```
opam-version: "2.0"
version: "2.0"
synopsis:
  "A toolset for describing and simulating StateChart-like state diagrams"
maintainer: ["Jocelyn Sérot <jocelyn.serot@uca.fr>"]
authors: ["Jocelyn Sérot <jocelyn.serot@uca.fr>"]
license: "MIT"
homepage: "http://github.com/jserot/rfsm"
bug-reports: "jocelyn.serot@uca.fr"
depends: [
  "dune" {>= "3.11"}
  "menhir" {>= "2.0"}
  "ocamlgraph" {>= "2.0"}
  "ppx_deriving" {>= "5.0"}
  "ocaml" {>= "5.0.0"}
  "odoc" {with-doc}
]
build: [
  ["dune" "subst"] {dev}
  [
    "dune"
    "build"
    "-p"
    name
    "-j"
    jobs
    "@install"
    "@runtest" {with-test}
    "@doc" {with-doc}
  ]
]
dev-repo: "git+https://github.com/jserot/rfsm.git"
```
There's no `src/old` subdir in the GH repo.
And i have no idea when and why the file `https://github.com/jserot/rfsm/archive/refs/tags/2.0.tar.gz` has been created. 
I don't even know how to access the directory ``https://github.com/jserot/rfsm/archive`, which is not listed on the `https://github.com/jserot/rfsm` page...

haochenx commented:
If your are using `dune-release` it will automatically generate a GitHub release (in this case it's https://github.com/jserot/rfsm/releases/tag/2.0) and create a opam-repo version of the opam file for you. If you examine the contents of this PR you will see: https://github.com/ocaml/opam-repository/pull/24879/files#diff-db25f21326a1cb2953ff36b1b8a95d96723b8f7a7c69bb8cd7a75d4a27dcce43R33

I believe the tarball is automatically generated by `dune-release`. You can try to have a clean local git worktree and run `dune-release` again after removing the GitHub release from GitHub's web-ui.

jserot commented:
Thanks for your help.

> If your are using dune-release it will automatically generate a GitHub release (in this case it's https://github.com/jserot/rfsm/releases/tag/2.0)

 I will delete the GH release.

> If you examine the contents of this PR you will see: https://github.com/ocaml/opam-repository/pull/24879/files#diff-db25f21326a1cb2953ff36b1b8a95d96723b8f7a7c69bb8cd7a75d4a27dcce43R33

I guess this file has also been automatically generated by `dune-release` isn't it ?

jserot commented:
Ok. I've deleted the 2.0 release on GH and cleaned my local git source tree.
Running `open publish .`, i get:
```
$ opam publish .

The following will be published:
  - rfsm version 2.0 with opam file at /Users/jserot/Dev/ml/rfsm/rfsm.opam
    archive at https://github.com/jserot/rfsm/archive/refs/tags/2.0.tar.gz
```
Why is the "archive" file `2.0.tar.gz` still mentioned here ? 
There's no directory  `https://github.com/jserot/rfsm/archive`

haochenx commented:
> There's no directory https://github.com/jserot/rfsm/archive

`https://github.com/jserot/rfsm/archive` is the prefix of `https://github.com/jserot/rfsm/archive/refs/tags/*` urls, which are [GitHub APIs](https://docs.github.com/en/repositories/working-with-files/using-files/downloading-source-code-archives#source-code-archive-urls) and is not a directory in your repository, so you do not have to worry too much about it as long as a git tag is created and pushed to GitHub :)

haochenx commented:
As it seems that you are using dune with your project, it's probably a good idea to stick with `dune-release` for the whole release process (at least you have a known working workflow), including submitting to opam-repo and do not use `opam publish` directly to avoid unintended overlapping of steps performed by each tool.

jserot commented:
Ok, let's try 'dune-release':
```
$ dune-release -v -v
dune-release: [INFO] dune-release 1.6.2 running
[-] Building source archive
dune-release: [EXEC:8480] ['git' 'rev-parse' '--git-dir']
dune-release: [EXEC:8485] ['git' '--git-dir' '.git' '--work-tree' './'
                           'status' '--porcelain']
dune-release: [EXEC:8490] ['git' 'rev-parse' '--git-dir']
dune-release: [EXEC:8495] ['git' '--git-dir' '.git' '--work-tree' './'
                           'describe' '--always' '--tags' 'HEAD']
dune-release: [EXEC:8500] ['git' 'rev-parse' '--git-dir']
dune-release: [EXEC:8505] ['git' '--git-dir' '.git' '--work-tree' './'
                           'describe' '--always' '--tags' 'HEAD']
dune-release: [EXEC:8510] ['git' 'rev-parse' '--git-dir']
dune-release: [EXEC:8515] ['git' 'rev-parse' '--git-dir']
dune-release: [EXEC:8520] ['git' '--git-dir' '.git' '--work-tree' './'
                           'describe' '--always' '--tags' 'HEAD']
dune-release: [EXEC:8525] ['git' 'rev-parse' '--git-dir']
dune-release: [EXEC:8530] ['git' 'rev-parse' '--git-dir']
dune-release: [EXEC:8535] ['git' '--git-dir' '.git' 'rev-parse' '--verify'
                           'refs/tags/2.0-16-g2e0ac9b']
dune-release: [EXEC:8540] ['git' '--git-dir' '.git' 'show' '-s'
                           '--format=%ct' '2.0-16-g2e0ac9b^0']
dune-release: [EXEC:8545] ['git' '--git-dir' '.git' 'clone' '--local' '.git'
                           '_build/rfsm-2.0-16-g2e0ac9b.build']
dune-release: [EXEC:8562] ['git' '--git-dir'
                           '_build/rfsm-2.0-16-g2e0ac9b.build/.git'
                           '--work-tree' '_build/rfsm-2.0-16-g2e0ac9b.build/'
                           'checkout' '--quiet' '-b'
                           'dune-release-dist-2.0-16-g2e0ac9b'
                           '2.0-16-g2e0ac9b']
dune-release: [EXEC:8568] ['dune' 'subst']
dune-release: [INFO] Archiving rfsm-2.0-16-g2e0ac9b/.gitignore
...
...
dune-release: [INFO] Archiving rfsm-2.0-16-g2e0ac9b/src/host/lib/vhdl_types.ml
dune-release: [EXEC:8584] ['git' 'rev-parse' '--git-dir']
dune-release: [EXEC:8589] ['git' '--git-dir' '.git' '--work-tree' './'
                           'describe' '--always' '--tags' 'HEAD']
dune-release: [EXEC:8594] ['git' 'rev-parse' '--git-dir']
dune-release: [EXEC:8600] ['bzip2']
[+] Wrote archive _build/rfsm-2.0-16-g2e0ac9b.tbz
dune-release: [EXEC:8620] ['tar' '-xjf' 'rfsm-2.0-16-g2e0ac9b.tbz']

[-] Performing lint for package rfsm in _build/rfsm-2.0-16-g2e0ac9b
[ OK ] File README is present.
[ OK ] File LICENSE is present.
[ OK ] File CHANGES is present.
[ OK ] File opam is present.
dune-release: [EXEC:8621] ['opam' '--version']
dune-release: [INFO] Parsing opam file rfsm.opam
dune-release: [EXEC:8640] ['true']
dune-release: [EXEC:8641] ['opam' 'lint' '-s' 'rfsm.opam']
[ OK ] lint opam file rfsm.opam.
[ OK ] opam field synopsis is present
[ OK ] opam fields homepage and dev-repo can be parsed by dune-release
[ OK ] Skipping doc field linting, no doc field found
[ OK ] lint of _build/rfsm-2.0-16-g2e0ac9b and package rfsm success

[-] Building package in _build/rfsm-2.0-16-g2e0ac9b
dune-release: [EXEC:8660] ['true']
dune-release: [EXEC:8663] ['dune' 'build' '-p' 'rfsm']
(cd _build/.sandbox/2e2e14acc7cd4e504bca0a88520c1a31/default && /Users/jserot/.opam/5.0.0/bin/menhir --unused-tokens --unused-precedence-levels src/host/lib/host_parser.mly src/guests/std/bin/guest_parser.mly --base src/guests/std/bin/parser --infer-read-reply src/guests/std/bin/parser__mock.mli.inferred)
Warning: 6 states have shift/reduce conflicts.
Warning: 28 shift/reduce conflicts were arbitrarily resolved.
[ OK ] package(s) build               

[-] Running package tests in _build/rfsm-2.0-16-g2e0ac9b
dune-release: [EXEC:9269] ['true']
dune-release: [EXEC:9272] ['dune' 'runtest' '-p' 'rfsm']
[ OK ] package(s) pass the tests  
dune-release: [EXEC:9281] ['git' 'rev-parse' '--git-dir']
dune-release: [EXEC:9286] ['git' '--git-dir' '.git' '--work-tree' './'
                           'describe' '--always' '--tags' 'HEAD']
dune-release: [EXEC:9291] ['git' 'rev-parse' '--git-dir']
dune-release: [EXEC:9296] ['git' 'rev-parse' '--git-dir']
dune-release: [EXEC:9301] ['git' '--git-dir' '.git' 'rev-parse' '--verify'
                           'HEAD^0']

[+] Distribution for rfsm 2.0-16-g2e0ac9b
[+] Commit 2e0ac9bc25244b4b92bec73f825def8fdc3e8971
[+] Archive _build/rfsm-2.0-16-g2e0ac9b.tbz
dune-release: [INFO] Parsing opam file rfsm.opam
[-] Skipping documentation publication for package rfsm: no doc field in rfsm.opam
[-] Publishing distribution
dune-release: [EXEC:9307] ['git' 'rev-parse' '--git-dir']
dune-release: [EXEC:9312] ['git' '--git-dir' '.git' '--work-tree' './'
                           'describe' '--always' '--tags' 'HEAD']
dune-release: [EXEC:9317] ['git' 'rev-parse' '--git-dir']
[-] Publishing to github
dune-release: [EXEC:9323] ['git' 'rev-parse' '--git-dir']
dune-release: [EXEC:9328] ['git' '--git-dir' '.git' '--work-tree' './'
                           'describe' '--always' '--tags' 'HEAD']
dune-release: [EXEC:9333] ['git' 'rev-parse' '--git-dir']
dune-release: [EXEC:9338] ['git' '--git-dir' '.git' 'rev-parse' '--verify'
                           'refs/tags/2.0-16-g2e0ac9b']
dune-release: [EXEC:9343] ['git' 'rev-parse' '--git-dir']
dune-release: [EXEC:9348] ['git' '--git-dir' '.git' 'rev-parse' '--verify'
                           'refs/tags/2.0-16-g2e0ac9b']
dune-release: [EXEC:9353] ['git' 'rev-parse' '--git-dir']
dune-release: [EXEC:9358] ['git' '--git-dir' '.git' '--work-tree' './'
                           'describe' '--always' '--tags' 'HEAD']
dune-release: [EXEC:9363] ['git' 'rev-parse' '--git-dir']
dune-release: [EXEC:9368] ['git' '--git-dir' '.git' 'rev-parse' '--verify'
                           '2.0-16-g2e0ac9b^0']
dune-release: [EXEC:9373] ['git' '--git-dir' '.git' 'ls-remote' '--quiet'
                           '--tags' 'https://github.com/jserot/rfsm.git'
                           '2.0-16-g2e0ac9b']
[?] Push tag 2.0-16-g2e0ac9b to git@github.com:jserot/rfsm.git? [Y/n]
Y
[-] Pushing tag 2.0-16-g2e0ac9b to git@github.com:jserot/rfsm.git
dune-release: [EXEC:9380] ['git' '--git-dir' '.git' 'push' '--force'
                           'git@github.com:jserot/rfsm.git'
                           '2.0-16-g2e0ac9b']
dune-release: [ERROR] Exit code 128 from command
    `git --git-dir .git push --force git@github.com:jserot/rfsm.git
       2.0-16-g2e0ac9b`:
  fatal: 2.0-16-g2e0ac9b cannot be resolved to branch.
  
  Pushing the tag failed, please push it manually and run the command again
```
What is going wrong now ?
